cfpb/consumerfinance.gov
# Add support for Elasticsearch-based filterable lists

Swaps our functionality for searching with FilterableLists to use Elasticsearch instead of postgres. 

---




## Additions

- Elasticsearch Index, Document, and Search classes. 
- Concept of "filterable_root" which defines the parent to which the filterable lists searches under
- "filterable_categories" added to the base FilterableMixin class to enable searching through only specific categories sitewide. 


## Changes

- No longer search using django's ORM. 


## How to test this PR

1. Visit any of our pages that leverage the FilterableList and search
2.  Compare results with production to see if there are any notable gaps


## Screenshots


## Notes and todos

- Setup Flag to enable toggling functionality
- Unit Tests
- Determine how we want to search the title - Do we want to use a wildcard to match or only match exact terms?


## Checklist



  - PR titles are used to generate the change log in [releases](../../releases); good ones make that easier to scan.
  - Consider prefixing, e.g., "Mega Menu: fix layout bug", or "Docs: Update Docker installation instructions".
  - [This repo’s docs](https://cfpb.github.io/consumerfinance.gov/) (edit the files in the `/docs` folder) – for basic, close-to-the-code docs on working with this repo
  - CFGOV/platform wiki on GHE – for internal CFPB developer guidance
  - CFPB/hubcap wiki on GHE – for internal CFPB design and content guidance

### Front-end testing



#### Browser testing

Visually tested in the following supported browsers:



#### Accessibility


#### Other
